File: assets/code-variants/ch12/25.py
class UserStoreTest(unittest.TestCase):
    def setUp(self):
        self.name_service = NameService()
        self.name_service.set("user1", "Donald Knuth")
        self.user_store = UserStore(self.name_service)

    # [... hundreds of lines of tests ...]

    def test_should_return_name_from_service(self):
        user = self.user_store.get("user1")
        self.assertEqual(user.name, "Donald Knuth")

File: assets/code-variants/ch12/25.java
private NameService nameService;
private UserStore userStore;

@Before
public void setUp() {
    nameService = new NameService();
    nameService.set("user1", "Donald Knuth");
    userStore = new UserStore(nameService);
}

// [... hundreds of lines of tests ...]

@Test
public void shouldReturnNameFromService() {
    UserDetails user = userStore.get("user1");
    assertThat(user.getName()).isEqualTo("Donald Knuth");
}
